src/ai/defaultai.cc

```diff
@@ -1497,15 +1497,15 @@ void DefaultAI::update_all_not_buildable_fields(const Time& gametime) {
 	// just counter
 	uint32_t checked_fields = 0;
 
-	while (maxchecks--) {
+	while ((maxchecks--) != 0u) {
```

Comment:
same here do the inner brackets effect the value of maxchecks that is evaluated as the value befor dekrement should be evaluated